### ISSUE TITLE
Datadog chart 2.19.2 generates invalid YAML

### ISSUE THREAD

**Issue by vladlosev** [DataDog/helm-charts#317]
**Describe what happened:**

Expanding the chart on a non-Windows system with APM enabled generates invalid configuration for the `trace-agent` container in the `datadog` DaemonSet:

```
MacBook:helm-charts vlad$ helm repo add datadog https://helm.datadoghq.com
"datadog" has been added to your repositories
MacBook:helm-charts vlad$ helm template datadog datadog/datadog --set=datadog.apm.enabled=true --version=2.19.2 | yq eval 'select(.kind == "DaemonSet") | .spec.template.spec.containers[] | select(.name == "trace-agent") | .volumeMounts' -
- name: config
  mountPath: /etc/datadog-agent
- name: logdatadog
  mountPath: /var/log/datadog
- name: tmpdir
  mountPath: /tmp
  readOnly: false
- name: dsdsocket
  mountPath: /var/run/datadog
- name: runtimesocketdir
  mountPath: /host/var/run
  mountPropagation: None
  readOnly: true
  readOnly: true
```

Note the duplicated `readOnly: true` line at the end, which is actually an invalid YAML. Even though `kubectl` will accept it, many validation tools will not.

It appears that changes in https://github.com/DataDog/helm-charts/pull/313/files#diff-345498eca0364ad56d08ede720269bf2efa8474f51e268a0a72444b5ef9008eeL75 caused this issue.

**Describe what you expected:**

Expected valid YAML output. To compare:
```
MacBook:helm-charts vlad$ helm template datadog datadog/datadog --set=datadog.apm.enabled=true --version=2.19.1 | yq eval 'select(.kind == "DaemonSet") | .spec.template.spec.containers[] | select(.name == "trace-agent") | .volumeMounts' -
- name: config
  mountPath: /etc/datadog-agent
- name: logdatadog
  mountPath: /var/log/datadog
- name: tmpdir
  mountPath: /tmp
  readOnly: false
- name: runtimesocketdir
  mountPath: /host/var/run
  mountPropagation: None
  readOnly: true
- name: dsdsocket
  mountPath: /var/run/datadog
  readOnly: true
  ```

**Steps to reproduce the issue:**

```
helm repo add datadog https://helm.datadoghq.com
helm template datadog datadog/datadog --set=datadog.apm.enabled=true --version=2.19.2
```

**Additional environment details (Operating System, Cloud provider, etc):**

OS: Linux, MacOS

**Comment by johnmartel** [comment]
In our case, this also prevents the `trace-agent` container from starting in a GKE environment. We systematically get the following error:

```text
Error response from daemon: invalid volume specification: '\\.\pipe\docker_engine:\\.\pipe\docker_engine:ro': invalid mount config for type "npipe": field ReadOnly must not be specified
```

**Comment by johnmartel** [comment]
An _interesting_ fact is that we can't seem to be able to rollback to 2.16.6 by simply modifying the version of the `helm_release` and reapplying the terraform changes. Currently using terraform 1.0.2 with the `hashicorp/helm` provider 2.2.0.

**Comment by jbfavre** [comment]
Also have the issue with 2.19.3 version of the chart

**Comment by johnmartel** [comment]
I think #318 fixes the issue and is part of 2.19.4

**Comment by vboulineau** [comment]
I think this transient issue is fixed now, feel free to re-open if you still face any issue.
